README.md

```diff
@@ -51,6 +56,46 @@ loosely based on the [ngx_http_auth_request_module] module for nginx.
         http-request deny if ! { var(txn.auth_response_successful) -m bool }
     ```
 
+### Parameters
+
+The scripts receive a list of params used to build the authentication request:
+
+* **Backend name**: is the name of an HAProxy backend. See the
+[Inner Workings](#inner-workings) section.
+* **Path to request**: the request URL sent to the auth-request backend.
+
+The following parameters are only available in the `auth-intercept` script:
+
+* **Method**: the HTTP method that should be used. Use an asterisk `*` to ask
+`auth-intercept` to copy the same method used by the client. `auth-request`
+uses the `HEAD` method.
+* **Headers to copy on Request**: a comma-separated list of [Lua Patterns] that
+should match the HTTP header names to copy from the client to the
+auth-intercept backend. Use a dash `-` to not copy any header.
+* **Headers to copy on Success**: a comma-separated list of [Lua Patterns] that
+should match the HTTP header names to copy from the auth-intercept backend to
+the protected backend server, if the auth-intercept backend respond with 2xx
+response code and the request succeed. All headers received from the
+auth-intercept will override headers with the same name provided by the client.
+Use `.*` to copy all headers, or use a dash `-` to not copy any header but,
+instead, create `req.auth_response_header.*` HAProxy variables. See the
+[Available Variables](#available-variables) section.
+* **Headers to copy on Failure**: a comma-separated list of [Lua Patterns] that
+should match the HTTP header names to copy from the auth-intercept backend to
+the client, if the request failed. `auth-intercept` will use the same HTTP
+method and body sent by the auth-intercept backend to respond to the client,
+closing the transaction. The protected backend server will not be used. Use
+`.*` to copy all headers. Use a dash `-` to not close the transaction and leave
+to the HAProxy configuration the task to deny the request based on the
+`txn.auth_response_successful` variable. Using a dash `-` will also create
+`req.auth_response_header.*` HAProxy variables. See the
+[Available Variables](#available-variables) section.
+
+A quick note about [Lua Patterns]: always escape the *magic characters* with a
+`%`, otherwise the match will not work as expected. For example, the pattern
+`x%-.*` is the correct way to filter only header names that starts with `x-`,
+and `x%-user` to only send the `x-user` HTTP header.
```

Comment:
I wonder if escaping of the hyphen (`-`) should be performed automatically, it's just *so* common in HTTP headers and the difference between the asterisk (`*`) should not be relevant for this use case.

Reply:
I also agree that escaped hyphen is handy and it's the best option on 90%+ of the use cases, however this isn't Lua pattern anymore and should be documented. Another option is to accept some kind of a glob pattern, escaping all of the magic chars and converting `*` to `.*` and `?` to `.`, how does it sound? I'm fine for all of them, I already have an automation that builds the parameter and can convert to what seems to the the best for the component.

ps - does it *hyphen* sound better than *dash* in english in this context? if so I'll convert all of them in the readme.

Reply:
> Another option is to accept some kind of a glob pattern, escaping all of the magic chars and converting `*` to `.*` and `?` to `.`, how does it sound?

That sounds good indeed. I would expect the common use-case to be that one wants to match all headers with a specific prefix and that would be possible when just supporting `*` and `?`. It's also well-known from shell globs.

> ps - does it _hyphen_ sound better than _dash_ in english in this context? if so I'll convert all of them in the readme.

Good question! I'm not a native speaker either. I did a quick search and found this: https://www.grammarly.com/blog/hyphens-and-dashes/. Basically a `-` *within* a single word is called a hyphen, while a `-` between parts of a sentence is a dash. In the case of an HTTP header “hyphen” would be more fitting.

Reply:
There is also the option to not copy the headers, it's not connecting letters or words, it's just a lonely ... dash. Or an hyphen? =) Ok taking care of the glob->lua pattern converter right now, I'll rebase everything into a single commit shortly.

Reply:
> There is also the option to not copy the headers, it's not connecting letters or words, it's just a lonely ... dash. Or an hyphen? =)

I believe that's a dash indeed! In the end I guess that the user will know what's meant. Especially since the character is also shown. Use whatever wording you are comfortable with :smile: